Sources/X509/CSR/CertificateSigningRequest.swift

```diff
@@ -145,6 +145,32 @@ public struct CertificateSigningRequest {
         self.signatureBytes = try DER.Serializer.serialized(element: ASN1BitString(self.signature))[...]
     }
 
+    /// Construct a CSR for a specific private key.
+    ///
+    /// This API can be used to construct a certificate signing request that can be passed to a certificate
+    /// authority. It will correctly generate a signature over the request.
```

Comment:
Let's add a note to point to the other initializer and indicate that this defaults the signature algorithm based on the private key.

Reply:
Good idea.